### PR TITLE
feat: add front matter variable to open in new tab

### PR DESCRIPTION
**Pull request by Avery2**
Similar to (#77), allows links in the navbar to open in a new tab. This time it is optional with front matter `open_new_tab`.

Example:
```
---
title: <i class="fab fa-github"></i>
external_url: https://github.com/Avery2
open_new_tab: true
weight: 99
---
```

### CODE REVIEW

**Comment by YoussefRaafatNasry** [comment]
I like this approach more, but I think that we should add it to other content that can have an external URL too, like projects and posts.

File: _includes/navbar.html [hunk 1]
```diff
@@ -51,7 +51,11 @@
           {%- assign url = html_page.url | relative_url -%}
         {%- endif -%}
 
-        <a class="nav-item nav-link {{ active_status }}" href="{{ url }}">{{ html_page.title }}</a>
+        {%- if html_page.open_new_tab -%}
+          <a class="nav-item nav-link {{ active_status }}" href="{{ url }}" target="_blank">{{ html_page.title }}</a>
+        {%- else -%}
+          <a class="nav-item nav-link {{ active_status }}" href="{{ url }}">{{ html_page.title }}</a>
+        {%- endif -%}
```
Comment (YoussefRaafatNasry) [inline _includes/navbar.html:58]: An inline if statement would be cleaner.

```suggestion
          <a class="nav-item nav-link {{ active_status }}" href="{{ url }}" {%- if html_page.open_new_tab -%} target="_blank" {%- endif -%}>{{ html_page.title }}</a>
```

**Review by YoussefRaafatNasry — CHANGES_REQUESTED** [review]

**Comment by YoussefRaafatNasry** [comment]
We can make this as a site-level configuration in `_config.yaml` to make it easier to standardize the behavior across site. What do you think?

**Comment by Avery2** [comment]
> We can make this as a site-level configuration in `_config.yml` to make it easier to standardize the behavior across site.
> What do you think?

Yeah, that sounds good. I'll try to do that.

**Comment by Avery2** [comment]
> We can make this as a site-level configuration in `_config.yml` to make it easier to standardize the behavior across site.

Right now there is just a setting `open_new_tab` in `_config.yml` that affects the default behavior of the links in the navbar. I have not yet been able to implement this in the projects or posts.

**Comment by Avery2** [comment]
Works for navbar (pages), projects, and posts. Users can edit the `open_new_tab` front matter in any of the corresponding files. They also can set site-wide defaults in the `_config.yml` (default value is false - so the default behavior will remain the same). Sorry that the git history is a little horrendous, I wasn't sure how to develop locally with the `remote_theme`.

**Comment by YoussefRaafatNasry** [comment]
@Avery2 the PR have irrelevant changes from your fork, consider separating them or closing the PR and creating a new one. 
Sorry for the late response.

**Comment by Avery2** [comment]
> @Avery2 the PR have irrelevant changes from your fork, consider separating them or closing the PR and creating a new one.
> Sorry for the late response.

My bad, I will change that.

**Comment by Avery2** [comment]
New PR at #82

**Pull request closed without merging**